test/mock/lib/request.js

```diff
@@ -135,6 +183,74 @@ Response.prototype.toBin = function() {
   return Buffer.concat(buffers);
 }
 
+CompressedResponse.prototype.toBin = function() {
+  var self = this;
+  var buffers = [];
+
+  // Serialize all the docs
+  var docs = this.uncompressedResponse.documents.map(function(x) {
+    return self.bson.serialize(x);
+  });
+
+  // Document total size
+  var uncompressedSize = 4 + 8 + 4 + 4; // OP_REPLY Header size
+  docs.forEach(function(x) {
+    uncompressedSize = uncompressedSize + x.length;
+  });
+
+  var dataToBeCompressedHeader = new Buffer(20);
+  var dataToBeCompressedBody = Buffer.concat(docs);
+
+  // Write response flags
+  writeInt32(dataToBeCompressedHeader, 0, this.uncompressedResponse.responseFlags)
+  writeInt64(dataToBeCompressedHeader, 4, this.uncompressedResponse.cursorId)
+  writeInt32(dataToBeCompressedHeader, 12, this.uncompressedResponse.startingFrom)
+  writeInt32(dataToBeCompressedHeader, 16, this.uncompressedResponse.numberReturned)
+
+  var dataToBeCompressed = Buffer.concat([dataToBeCompressedHeader, dataToBeCompressedBody])
+
+  // Compress the data
+  var compressorFunctions = [
```

Comment:
less important because its in the testing, but again we are constructing this array _every time_ you compress something